Silsla 13/MARC manipulation for SILSLA-13

aprigge opened the pull request:
MARC manipulations for [SILSLA_13](https://jira.library.ucla.edu/browse/SILSLA-13). It seems that changing the value of Field.tag removes the old field, so there is no need to remove it.

aprigge committed:
SILSLA-13: Initial MARC modifications.

aprigge committed:
SILSLA-13: Initial MARC manipulations.

aprigge committed:
SILSLA-13: MARC field manipulations for SILSLA.

aprigge committed:
SILSLA-13: Removed print statement.

akohler commented on vanguard.py:
```diff
@@ -0,0 +1,49 @@
+	for old_field in range(500, 600):
+		if old_field != 590:
+			for fld in record.get_fields(old_field):
+				for sfld in fld.get_subfields('5'):
```
The $5 is not supposed to be repeated
https://www.loc.gov/marc/bibliographic/bd700.html (and others)
so you can simplify this by just testing the $5 directly like `if fld['5'].startswith('CLU')`

That would simplify most of these methods.

akohler commented on vanguard.py:
```diff
@@ -0,0 +1,49 @@
+					if sfld.startswith('CLU'):
+						fld.tag = '590'
+						record.add_ordered_field(fld)
+	return record
```
Since the calling code is not using the return value, and we're passing a mutable object (`record`) by reference, changes made to `record` within the function are automatically available to the caller.  So these `return` statements can be removed.

akohler commented on vanguard.py:
```diff
@@ -0,0 +1,49 @@
+						record.add_ordered_field(fld)
+	return record
+
+def change_CLU(record):
```
I think this function should be renamed.  It does the same basic thing as `modify_5xx()`, so should have a similar name.
Since there is a lot of duplicated code, maybe use the `field_mapping` approach: add a helper function which adds the 99 relevant entries from the range (500-599, minus 590) to `field_mapping`, then you could use `change_CLU()` and remove `modify_5xx()`.

akohler commented on vanguard.py:
```diff
@@ -0,0 +1,49 @@
+				record.remove_field(fld)			
+	return record
+
+def clean_record(file_name=None, out_file=None):
```
What happens if MARCReader or MARCWriter gets `None` here?  Are you expecting that the program or main method which calls `clean_record()` will ensure that file names are provided?

Also, I recommend changing `file_name` to `in_file` or similar, to show the relationship to `out_file`.

akohler reviewed (changes requested):
@aprigge I think this works, but can be improved.  Please take a look at my suggestions and let me know if any questions or counter-proposals :)

Thanks --Andy

aprigge committed:
SILSLA_13: Refactored get_5xx_fields, change_CLU, and reader and writer.

aprigge commented:
@akohler , thank you for the thoughtful comments! I don't have a counter offer, so I've implemented the changes as you described. I also changed the reader/writer portion to match the scp_bib_update script. Let me know if you see any other needed changes. Thanks, Ashton

aprigge committed:
SILSLA_13: Removed print statement.

akohler commented:
@aprigge Thanks for the changes, looks good.  Merged!
